File: app/javascript/services/core-utils.js
export function setUser(user){
    console.log(user)
}

export class base36{
    ALPHABET = "abcdefghijklmnopqrstuvexyz1234567890"
    R_ALPHABET = {}
    _BASE = this.ALPHABET.length
    constructor(){
        for(let i = 0; i < this._BASE; i += 1){
            this.R_ALPHABET[this.ALPHABET[i]] = i;
        }
    }

    base36encode(base10num){
        let num= base10num
        let r = []
        while (num > 0){
            num = Math.floor(num/this._BASE)
            let rem = num%this._BASE
            r.push(this.ALPHABET[rem])
        }
        return r.join("")
    }

    base36decode(str){
        let r = 0
        let n = str.length
        let idx = str.length - 1
        while (idx != -1){
            r += this.R_ALPHABET[str[idx]] * (this._BASE ** (n - idx - 1))
            idx -= 1
        }
        return r;
    }   
}



export function handleInputChange(event) {
    const target = event.target;
    //console.log(target)
    const dataAttrs = target.dataset
    if ("id" in dataAttrs){
      const value = target.value//target.type === 'checkbox' ? target.checked : target.value;
      const name = target.name
      this.setState({
        [name]: value
      });
    }
  }

File: app/controllers/services/core_utils.rb


def url_generator
    url = params[:url]
    domain = ""
    path = params[:path]
    
end

def url_parser
    url = params[:url]
end
class Base36
    @ALPHABET = "abcdefghijklmnopqrstuvexyz1234567890"
    @R_ALPHABET = {}
    @_BASE = @ALPHABET.length
    def initialize
        for i in 0...@_BASE
            @R_ALPHABET[@ALPHABET[i]] = i;
        end
    end

    def base36encode(base10num){
        base10num = params[:base10num]
        num= base10num
        r = []
        while num > 0 do
            num = Math.floor(num/@_BASE)
            rem = num%this._BASE
            r.append(@ALPHABET[rem])
        end
        return r.join("")
    end

    def base36decode
        r = 0
        str = params[:str]
        n = str.length
        idx = str.length - 1
        while idx != -1 do
            r += @R_ALPHABET[str[idx]] * (@_BASE ** (n - idx - 1))
            idx -= 1
        end
        return r;
    end
}